### PR TITLE
ovl_Obj_Dowsing ok

### PR DESCRIPTION
**Pull request by shawlucas**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

ok

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.c [hunk 1]
```diff
@@ -10,11 +10,12 @@
 
 #define THIS ((ObjDowsing*)thisx)
 
+s32 ObjDowsing_GetFlag(ObjDowsing* this, GlobalContext* globalCtx);
+s32 func_80B23DD0(ObjDowsing* this, GlobalContext* globalCtx);
 void ObjDowsing_Init(Actor* thisx, GlobalContext* globalCtx);
 void ObjDowsing_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void ObjDowsing_Update(Actor* thisx, GlobalContext* globalCtx);
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.c:17]:
```suggestion
void ObjDowsing_Init(Actor* thisx, GlobalContext* globalCtx);
void ObjDowsing_Destroy(Actor* thisx, GlobalContext* globalCtx);
void ObjDowsing_Update(Actor* thisx, GlobalContext* globalCtx);

s32 ObjDowsing_GetFlag(ObjDowsing* this, GlobalContext* globalCtx);
s32 func_80B23DD0(ObjDowsing* this, GlobalContext* globalCtx);
```
We always put the other functions in the actor *below* the external ones.

File: src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.c [hunk 2]
```diff
@@ -27,14 +28,42 @@ const ActorInit Obj_Dowsing_InitVars = {
...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Dowsing/func_80B23DD0.s")
+s32 func_80B23DD0(ObjDowsing* this, GlobalContext* globalCtx) {
+    if (ObjDowsing_GetFlag(this, globalCtx)) {
+        Actor_MarkForDeath(&this->actor);
+        return 1;
+    }
+    return 0;
+}
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.c:52]:
```suggestion
s32 func_80B23DD0(ObjDowsing* this, GlobalContext* globalCtx) {
    if (ObjDowsing_GetFlag(this, globalCtx)) {
        Actor_MarkForDeath(&this->actor);
        return true;
    }
    return false;
}
```
This is used as a boolean. Can also probably call it something like `CheckValidSpawn`

File: src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.h [hunk 3]
```diff
@@ -3,8 +3,18 @@
 
 #include "global.h"
 
+#define DOWSING_GET_TYPE(this) ((this)->actor.params >> 7)
+#define DOWSING_GET_FLAG(this) ((this)->actor.params & 0x7F)
+
 struct ObjDowsing;
 
+typedef enum {
+    /* 0 */ DOWSING_UNK_00,
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.h:12]: Is 0 used? The enum might just start at 1.

File: src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.h [hunk 4]
```diff
@@ -3,8 +3,18 @@
 
 #include "global.h"
 
+#define DOWSING_GET_TYPE(this) ((this)->actor.params >> 7)
+#define DOWSING_GET_FLAG(this) ((this)->actor.params & 0x7F)
```
Comment (hensldm) [inline src/overlays/actors/ovl_Obj_Dowsing/z_obj_dowsing.h:7]:
```suggestion
#define DOWSING_GET_TYPE(thisx) ((thisx)->params >> 7)
#define DOWSING_GET_FLAG(thisx) ((thisx)->params & 0x7F)
```
Current standard is too pass in the base actor pointer. You will have to adjust how the macros are used in `ObjDowsing_GetFlag`.

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**